test/integration/etcd/etcd_storage_path_test.go

```diff
@@ -214,6 +214,13 @@ var etcdStorageData = map[schema.GroupVersionResource]struct {
 	},
 	// --
 
+	// k8s.io/kubernetes/pkg/apis/networking/v1
+	gvr("networking.k8s.io", "v1", "networkpolicies"): {
+		stub:             `{"metadata": {"name": "np1"}, "spec": {"podSelector": {"matchLabels": {"e": "f"}}}}`,
+		expectedEtcdPath: "/registry/networkpolicies/etcdstoragepathtestnamespace/np1",
+	},
```

Comment:
@liggitt This may be related to the cache?

```
I0515 18:07:41.270516     973 storage_rbac.go:257] created rolebinding.rbac.authorization.k8s.io/system:controller:token-cleaner in kube-system
I0515 18:07:41.310316     973 storage_rbac.go:257] created rolebinding.rbac.authorization.k8s.io/system:controller:bootstrap-signer in kube-public
E0515 18:07:42.994666     973 watcher.go:247] failed to prepare current and previous objects: v1.NetworkPolicy is not suitable for converting to {{"extensions" "__internal"} [{"extensions" ""} {"extensions" ""}]}
W0515 18:07:42.996082     973 reflector.go:323] k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/storage/cacher.go:215: watch of *extensions.NetworkPolicy ended with: v1.NetworkPolicy is not suitable for converting to {{"extensions" "__internal"} [{"extensions" ""} {"extensions" ""}]}
E0515 18:07:59.659734     973 watcher.go:247] failed to prepare current and previous objects: v1beta1.NetworkPolicy is not suitable for converting to {{"networking.k8s.io" "__internal"} [{"networking.k8s.io" ""} {"networking.k8s.io" ""}]}
W0515 18:07:59.659912     973 reflector.go:323] k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/storage/cacher.go:215: watch of *networking.NetworkPolicy ended with: v1beta1.NetworkPolicy is not suitable for converting to {{"networking.k8s.io" "__internal"} [{"networking.k8s.io" ""} {"networking.k8s.io" ""}]}
--- PASS: TestEtcdStoragePath (26.25s)
PASS
ok  	k8s.io/kubernetes/test/integration/etcd	26.389s
+++ [0515 18:08:04] Cleaning up etcd
+++ [0515 18:08:04] Integration test cleanup complete
```

I added some debug logic and changed the paths:

```
GVK: schema.GroupVersionKind{Group:"networking.k8s.io", Version:"v1", Kind:"NetworkPolicy"}
GVR: schema.GroupVersionResource{Group:"networking.k8s.io", Version:"v1", Resource:"networkpolicies"}
expected: schema.GroupVersionKind{Group:"networking.k8s.io", Version:"v1", Kind:"NetworkPolicy"}
actual: schema.GroupVersionKind{Group:"networking.k8s.io", Version:"v1", Kind:"NetworkPolicy"}
input: &etcd.metaObject{Kind:"", APIVersion:"", Metadata:struct { Name string "json:\"name,omitempty\" protobuf:\"bytes,1,opt,name=name\""; Namespace string "json:\"namespace,omitempty\" protobuf:\"bytes,2,opt,name=namespace\"" }{Name:"np2", Namespace:""}}
output: &etcd.metaObject{Kind:"NetworkPolicy", APIVersion:"networking.k8s.io/v1", Metadata:struct { Name string "json:\"name,omitempty\" protobuf:\"bytes,1,opt,name=name\""; Namespace string "json:\"namespace,omitempty\" protobuf:\"bytes,2,opt,name=namespace\"" }{Name:"np2", Namespace:"etcdstoragepathtestnamespace"}}
testdata: struct { stub string; prerequisites []etcd.prerequisite; expectedEtcdPath string; expectedGVK *schema.GroupVersionKind }{stub:"{\"metadata\": {\"name\": \"np2\"}, \"spec\": {\"podSelector\": {\"matchLabels\": {\"e\": \"f\"}}}}", prerequisites:[]etcd.prerequisite(nil), expectedEtcdPath:"/registry/networkpolicies/etcdstoragepathtestnamespace/np2", expectedGVK:(*schema.GroupVersionKind)(nil)}

GVK: schema.GroupVersionKind{Group:"extensions", Version:"v1beta1", Kind:"NetworkPolicy"}
GVR: schema.GroupVersionResource{Group:"extensions", Version:"v1beta1", Resource:"networkpolicies"}
expected: schema.GroupVersionKind{Group:"extensions", Version:"v1beta1", Kind:"NetworkPolicy"}
actual: schema.GroupVersionKind{Group:"extensions", Version:"v1beta1", Kind:"NetworkPolicy"}
input: &etcd.metaObject{Kind:"", APIVersion:"", Metadata:struct { Name string "json:\"name,omitempty\" protobuf:\"bytes,1,opt,name=name\""; Namespace string "json:\"namespace,omitempty\" protobuf:\"bytes,2,opt,name=namespace\"" }{Name:"np1", Namespace:""}}
output: &etcd.metaObject{Kind:"NetworkPolicy", APIVersion:"extensions/v1beta1", Metadata:struct { Name string "json:\"name,omitempty\" protobuf:\"bytes,1,opt,name=name\""; Namespace string "json:\"namespace,omitempty\" protobuf:\"bytes,2,opt,name=namespace\"" }{Name:"np1", Namespace:"etcdstoragepathtestnamespace"}}
testdata: struct { stub string; prerequisites []etcd.prerequisite; expectedEtcdPath string; expectedGVK *schema.GroupVersionKind }{stub:"{\"metadata\": {\"name\": \"np1\"}, \"spec\": {\"podSelector\": {\"matchLabels\": {\"e\": \"f\"}}}}", prerequisites:[]etcd.prerequisite(nil), expectedEtcdPath:"/registry/networkpolicies/etcdstoragepathtestnamespace/np1", expectedGVK:(*schema.GroupVersionKind)(nil)}
```

Do we need to set them as `CohabitatingResources`?